src/Microsoft.OData.Edm/ExtensionMethods/TargetPathHelper.cs

```diff
@@ -0,0 +1,235 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TargetPathHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.OData.Edm.Vocabularies;
+
+    /// <summary>
+    /// Target path helper.
+    /// </summary>
+    internal static class TargetPathHelper
+    {
+        /// <summary>
+        /// Get the Segments from a target path string as enumeration of Edm elements.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="targetSegments">Segments for the target path..</param>
+        /// <param name="caseInsensitive">Property name case-insensitive or not.</param>
+        /// <returns>The created enumeration of <see cref="IEdmElement"/>.</returns>
+        public static IEnumerable<IEdmElement> GetTargetSegments(this IEdmModel model, string[] targetSegments, bool caseInsensitive)
+        {
+            EdmUtil.CheckArgumentNull(model, nameof(model));
+            EdmUtil.CheckArgumentNull(targetSegments, nameof(targetSegments));
+
+            IEdmEntityContainer container = model.FindEntityContainer(targetSegments[0]);

```

Comment:
>FindEntityContainer [](http://example.com/codeflow?start=50&length=19)

Since you enable 'caseinsensitive", does 'FindEntityContainer" support caseInsensitive?

Reply:
We don't have a `FindEntityContainer` overload that accepts the `caseInsensitive` parameter.